src/blog/blog.controller.ts

```diff
@@ -95,8 +112,78 @@ export class BlogController {
         const blog = await blogRepository.findOne({ where: { id }, relations: ['author'] });
         if (!blog) return res.status(404).json({ message: 'Blog not found' });
 
-        await blogRepository.remove(blog); // Remove the blog
+        await blogRepository.remove(blog); 
 
         return res.status(200).json({ message: 'Blog deleted successfully' });
     }
-}
+
+    static async likeBlog(req: PrivateRequest, res: Response) {
```

Comment:
please save record only when user likes the blog. This record is used to identify which user liked particular post. when user clicks 'thumbs up' button, you will check whether user already like this post. If he didn't like yet (no record of user and post), then you insert one, if this record already exists, you delete this record (which means he decided to unlike post)

Reply:
removed dislike